php-frontend/src/main/java/org/sonar/php/metrics/CpdVisitor.java

```diff
@@ -132,11 +134,14 @@ private void storeCpdTokensInCache(PhpFile file, @Nullable CacheContext cacheCon
         CpdSerializationInput input = new CpdSerializationInput(cpdTokens, cacheContext.pluginVersion());
         SerializationResult serializationResult = CpdSerializer.toBinary(input);
 
-        String dataKey = CACHE_DATA_PREFIX + file.key();
-        writeCache.writeBytes(dataKey, serializationResult.data());
+        String hash = HashProvider.hash(file);
+        if(StringUtils.isNotBlank(hash)) {
```

Comment:
Why do you check for "isNotBlank" here compared to null check as previously?

Reply:
Good catch. Null check is good enough in both cases.